akka-stream/src/main/scala/akka/stream/impl/BoundedSourceQueue.scala

```diff
@@ -40,6 +43,7 @@ import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, Out
 
     val state = new AtomicReference[State](Running)
     val queue = new AbstractBoundedNodeQueue[T](bufferSize) {}
+    val awaitingSubscribers = new ConcurrentLinkedQueue[Promise[akka.Done]]()
```

Comment:
I'm not sure it is ok to use a Java queue here. Alternatively, I can use List + CAS for safe concurrent updates of the subscribers list. Could you, please, share your thoughts on the best suitable approach from the project perspective.